pkg/authorization/workspace_content_authorizer.go

```diff
@@ -92,6 +92,19 @@ func (a *workspaceContentAuthorizer) Authorize(ctx context.Context, attr authori
 	isServiceAccountFromRootCluster := subjectClusters[tenancyv1alpha1.RootCluster]
 	isServiceAccountFromCluster := subjectClusters[cluster.Name]
 
+	if IsDeepSubjectAccessReviewFrom(ctx, attr) {
+		attr := deepCopyAttributes(attr)
+		// this is a deep SAR request, we have to skip the checks here and delegate to the subsequent authorizer.
+		if isAuthenticated && !isUser && !isServiceAccountFromCluster {
+			// anonymize service accounts from other workspaces
+			attr.User = &user.DefaultInfo{
+				Name:   "system:anonymous",
+				Groups: []string{"system:authenticated"},
```

Comment:
@sttts i wonder if we should add `system:kcp:clusterworkspace:access` to both service accounts, and users here. There might be bindings relying on it.

Reply:
Am not 100% clear what is best. I tend to not add the access group. The caller of the SAR can add the group if it makes sense.

Reply:
makes sense